functional_hw/functional.py

```diff
@@ -0,0 +1,202 @@
+    """
+    filtered_values = list(filter(conditional_fun, values))
+
+    if not filtered_values:
+        raise IndexError("list with values passed by the filter has zero length")
+
+    result = filtered_values[0]
+    for next_val in filtered_values[1:]:
+        result = apply_fun(result, next_val)
+
+    return result
+
+
+def func_chain(*args):
+    """
+    func_chain takes functions and return one new function that includes all taken functions.
+    This new function applies all taken functions in the given order to given values.
+
+    [!] Functions apply in order that they were written.
+    [!] Functions should return application results of them.
+
+    New function takes some values (*values)
+    and returns list of modified values or one value (not listed) depended on number of values in result.
+
+    :param args: functions.
+    :return: new function includes taken functions.
+    """
+
+    def new_funct(*values):
+        for fun in args:
+            values = list(map(fun, values))
+        if len(values) == 1:
+            return values[0]
+        return values
+
+    return new_funct
+
+
+def sequential_map_2(*args):
+    """
+    [This function have the same functionality as sequential_map, but utilizes func_chain function in it.]
+
+    sequential_map_2 takes several functions, and container of values to which these functions will be applied
+    and return list of results after application all functions to values container.
+
+    [!] Functions should return application results of them.
+    [!] Container of values should always be written last.
+    [!] Functions apply in order that they were written.
+
+    :param args: functions and container of values (last!).
+    :return: list with modified values from container.
+    """
+
+    *functions, values = args
+    map_fun = func_chain(*functions)
+
+    return map_fun(*values)
+
+
+def multiple_partial(*functions, **params):
+    """
+    multiple_partial takes functions (*functions) and modifies them with setting their parameters (*params) by default.
+
+    multiple_partial includes func_factory function to create new independent functions with set parameters.
+
+    [!] If some modified function is used with new parameters which differ from set with multiple_partial,
+    modified function will use new parameters.
+
+    :param functions: functions to modify.
+    :param params: parameters to set default.
+    :return: list of modified functions.
+    """
+
+    def func_factory(funct_to_change):
+        def partial_fun(*args, **kwargs):
+            params.update(kwargs)
+            return funct_to_change(*args, **params)
+
+        return partial_fun
+
+    list_of_functions = []
+
+    for fun in functions:
+        new_fun = func_factory(fun)
+        list_of_functions.append(new_fun)
+
+    return list_of_functions
+
+
+def super_print(*args, sep=' ', end='\n', file=None):
```

Comment:
```suggestion
def super_print(*args, sep=' ', end='\n', file=sys.stdout):
```
В встроенном принте так